src/hooks/useButtonAnimation.ts

```diff
@@ -15,7 +15,7 @@ const useButtonAnimation = (iterateValues: number[], transitionDelay: number): P
     if (iterateValues.length !== 0) {
       if (!iterateValues.includes(current) && update) setCurrent(iterateValues[0])
       timeout = setTimeout(() => {
-        if (update && isMounted) {
+        if (update && isMounted && transitionDelay !== -1) {
```

Comment:
I think there is another fix to make, but we should test it locally, in
`src/Widgets/PaymentPlans/index.tsx` 

There is this if : 

```
const realTransitionTime = () => {
    if (isTransitionSpecified) {
      return transitionDelay ?? DEFAULT_TRANSITION_TIME
    }
    if (isSuggestedPaymentPlanSpecified) {
      return VERY_LONG_TIME_IN_MS
    }
    return DEFAULT_TRANSITION_TIME
  }
  ```
  
  In this case, if `isSuggestedPaymentPlanSpecified` there is no way to cancel the loop, what if a merchant wants to highlight a specific fee plan but not use the loop feature ? I think we should also add the possibility to provide `isSuggestedPaymentPlanSpecified` AND `transitionDelay = -1` 
  
   like 
   
   ```
     const realTransitionTime = () => {
    if (isTransitionSpecified) {
      return transitionDelay ?? DEFAULT_TRANSITION_TIME
    }
    if (isSuggestedPaymentPlanSpecified) {
      return transitionDelay === -1 ? -1 : VERY_LONG_TIME_IN_MS
    }
    return DEFAULT_TRANSITION_TIME
  }
  ```
  
  WDYT ?

Reply:
Oh wait, but if we do this, then `isTransitionSpecified` would be true and we would never go into the `if (isSuggestedPaymentPlanSpecified) {`

Reply:
Nevermind it's already working as you fixed it :) 
![Capture d’écran 2023-10-04 à 15 17 19](https://github.com/alma/widgets/assets/33830009/17acafd2-8866-40d8-a90a-3922d255042f)
<img width="462" alt="Capture d’écran 2023-10-04 à 15 17 23" src="https://github.com/alma/widgets/assets/33830009/2fadd74c-b843-4ec7-99f1-cf4154226e13">
![Capture d’écran 2023-10-04 à 15 17 33](https://github.com/alma/widgets/assets/33830009/d5bd00f2-4cb6-4a40-8326-8571e9edf1be)
<img width="461" alt="Capture d’écran 2023-10-04 à 15 17 36" src="https://github.com/alma/widgets/assets/33830009/cfec5dfc-9ea2-4fe1-ae77-2172e8a354a3">